xdsl/dialects/builtin.py

```diff
@@ -1163,6 +1163,8 @@ class UnregisteredOp(IRDLOperation, ABC):
     res: VarOpResult = var_result_def()
     regs: VarRegion = var_region_def()
 
+    traits = frozenset()
```

Comment:
When I remove this the tests crash...

Reply:
Which test is crashing when you do that?
That seems weird

Reply:
I'll try without it and comment here

Reply:
```
$ "cd" "/Users/sasha/Developer/xdslproject/xdsl/docs/Toy"
$ ":" "RUN: at line 1"
$ "python" "-m" "toy" "/Users/sasha/Developer/xdslproject/xdsl/docs/Toy/examples/interpret.toy" "--emit=toy"
$ "filecheck" "/Users/sasha/Developer/xdslproject/xdsl/docs/Toy/examples/interpret.toy"
$ ":" "RUN: at line 2"
$ "python" "-m" "toy" "/Users/sasha/Developer/xdslproject/xdsl/docs/Toy/examples/interpret.toy" "--emit=toy-opt"
# command stderr:
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/sasha/Developer/xdslproject/xdsl/docs/Toy/toy/__main__.py", line 131, in <module>
    main(args.source, args.emit, args.ir, args.accelerate, args.print_generic)
  File "/Users/sasha/Developer/xdslproject/xdsl/docs/Toy/toy/__main__.py", line 80, in main
    transform(ctx, module_op, target=emit, accelerate=accelerate)
  File "/Users/sasha/Developer/xdslproject/xdsl/docs/Toy/toy/compiler.py", line 73, in transform
    CanonicalizationPass().apply(ctx, module_op)
  File "/Users/sasha/Developer/xdslproject/xdsl/xdsl/transforms/canonicalize/canonicalize_pass.py", line 45, in apply
    patterns = [
               ^
  File "/Users/sasha/Developer/xdslproject/xdsl/xdsl/transforms/canonicalize/canonicalize_pass.py", line 48, in <listcomp>
    if (trait := ctx_op.get_trait(HasCanonicalisationPatternsTrait)) is not None
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sasha/Developer/xdslproject/xdsl/xdsl/ir/core.py", line 915, in get_trait
    for t in cls.traits:
             ^^^^^^^^^^
AttributeError: type object 'UnregisteredOp' has no attribute 'traits'

error: command failed with exit status: 1
$ "filecheck" "/Users/sasha/Developer/xdslproject/xdsl/docs/Toy/examples/interpret.toy"
# command output:
CHECK: FileCheck error: '-' is empty.
FileCheck command line: /Users/sasha/Developer/xdslproject/xdsl/docs/Toy/examples/interpret.toy

error: command failed with exit status: 2
```